llmfoundry/data/finetuning/dataloader.py

```diff
@@ -145,6 +148,21 @@ def build_finetuning_dataloader(cfg: DictConfig, tokenizer: Tokenizer,
         )
 
     else:
+        if '://' in cfg.dataset.hf_name:
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                supported_extensions = ['jsonl', 'csv', 'parquet']
+                for extension in supported_extensions:
+                    name = f'{cfg.dataset.hf_name}/{cfg.dataset.split}.{extension}'
+                    location = f'{tmp_dir}/{name}'
+                    try:
+                        get_file(cfg.dataset.hf_name, location)
+                    except FileNotFoundError as e:
+                        if extension == supported_extensions[-1]:
+                            raise e
+                        continue
+                    cfg.dataset.hf_name = extension
+                    cfg.dataset.hf_kwargs['data_dir'] = tmp_dir
```

Comment:
this `tmp-dir` will be destroyed outside of the context manager

Reply:
Is it then better to make a non-temporary directory, or have the loading stay happen in the context manager and duplicate that line in an `else` block?

Reply:
The second option should work fine. I think the only reason you would not do it is that it is *mildly* harder to read. But perfectly acceptable IMHO.